src/main/java/de/tum/in/www1/artemis/web/rest/CourseResource.java

```diff
@@ -366,6 +365,80 @@ private void checkIfGroupsExists(Course course) {
         return ResponseUtil.wrapOrNotFound(Optional.ofNullable(course));
     }
 
+    /**
+     * GET  /courses/:id/with-exercises-and-relevant-participations
+     *
+     * Get the "id" course, with text and modelling exercises and their participations
+     *
+     * @param id the id of the course to retrieve
+     * @return the ResponseEntity with status 200 (OK) and with body the course, or with status 404 (Not Found)
+     */
+    @GetMapping("/courses/{id}/with-exercises-and-relevant-participations")
+    @PreAuthorize("hasAnyRole('TA', 'INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<Course> getCourseWithExercisesAndRelevantParticipations(@PathVariable Long id) {
+        log.debug("REST request to get Course with exercises and relevant participations : {}", id);
+        Course course = courseService.findOneWithExercises(id);
+
+        if (!userHasPermission(course)) return forbidden();
+
+        Set<Exercise> interestingExercises = course.getExercises().stream()
+            .filter(exercise -> exercise instanceof TextExercise || exercise instanceof ModelingExercise)
+            .collect(Collectors.toSet());
+
+        course.setExercises(interestingExercises);
+
+        List<Participation> participations = this.participationService.findByCourseIdWithRelevantResults(course.getId());
+
+        for (Exercise exercise : course.getExercises()) {
+            Stream<Participation> participationsForExercise = participations.stream()
+                .filter(participation -> participation.getExercise().getId().equals(exercise.getId()));
+
+            exercise.setParticipations(participationsForExercise.collect(Collectors.toSet()));
+        }
+
+        return ResponseUtil.wrapOrNotFound(Optional.of(course));
+    }
+
+    /**
+     * GET /courses/:id/stats-for-instructor-dashboard
+     * <p>
+     * A collection of useful statistics for the instructor course dashboard, including:
+     * - number of students
+     * - number of instructors
+     * - number of submissions
+     * - number of assessments
+     * - number of complaints
+     * - number of open complaints
+     *
+     * @param courseId the id of the course to retrieve
+     * @return data about a course including all exercises, plus some data for the tutor
+     * as tutor status for assessment
+     */
+    @GetMapping("/courses/{courseId}/stats-for-instructor-dashboard")
+    @PreAuthorize("hasAnyRole('INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<JsonNode> getStatsForInstructorDashboard(@PathVariable Long courseId) {
```

Comment:
getStatsForInstructorDashboard is a custom JSON response. is an example response documented somewhere? it could be useful to add something like that to the javadoc

Reply:
What about creating a DTO Class for that. Any thought on this topic from @krusche?

Reply:
A DTO would be fine from my point of view (as long as we do not create DTOs for normal domain objects)

Reply:
Done